sparrow-py/pysrc/sparrow_py/expr.py

```diff
@@ -314,8 +317,19 @@ def _aggregation(op: str, input: Expr, window: Optional[Window]) -> Expr:
     if window is None:
         return Expr.call(op, input, None, None)
     elif isinstance(window, SinceWindow):
-        return Expr.call(op, input, window._predicate, 1)
+        return Expr.call(op, input, window._predicate, None)
     elif isinstance(window, SlidingWindow):
         return Expr.call(op, input, window._predicate, window._duration)
     else:
         raise ValueError(f"Unknown window type {window!r}")
+
+def _collect(input: Expr, max: Optional[int], window: Optional[Window]) -> Expr:
```

Comment:
Can generalize the helper. It would be easiest if the `max` argument came *after* the window, but we can make it work:

```
def _aggregation(op: str, input: Expr, window: Optional[Window], *args: Optional[Arg]) -> Expr:
  """
  Create an aggregation.

  Parameters
  ----------
  op : str
      The operation to create.
  input : Expr
      The input to the expression.
  window : Optional[Window]
    The window to use for the aggregation.
  *args : Optional[Arg]
      Additional arguments to provide before `input` and the flattened window.

  Returns
  -------
  The resulting expression.
  """
  if window is None:
    # Note: things would be easier if we had a more normal order, which
    # we could do as part of "aligning" Sparrow signatures to the new direction.
    return Expr.call(op, *args, input, None, None)
  ...